spec/controllers/downloads_controller_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+describe DownloadsController do
+  describe '#show' do
+    let(:user) { FactoryGirl.create(:admin) }
+    let(:file_set) do
+      FactoryGirl.create(:file_set, user: user, content: File.open(fixture_path + '/files/video.mp4'))
+    end
+
+    context "when the user has access" do
+      before { sign_in user }
+
+      it 'sends the original file' do
+        get :show, :id => file_set.id
+        expect(response.body).to eq file_set.original_file.content
+      end
+
+      it 'sends the derivative' do
```

Comment:
Yes, that's a great idea. I suspect this is relevant to what that FileSets working group is tackling. So if @remerjohnson joins it, or one of us, we can track whether upstream support might be coming for service files, etc.